spec.bs

```diff
@@ -1260,8 +1261,24 @@ an [=ordered map=] whose [=map/keys=] are [=strings=] and whose [=map/values=] a
     1. Set |global|'s
       [=InterestGroupBiddingScriptRunnerGlobalScope/is component auction=] to |isComponentAuction|.
     1. Set |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/interest group=] to |ig|.
-    1. Return the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
+    1. Let |result| be the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
       |arguments|, and |timeout|.
+    1. If |result| is a [=throw completion=] with \[[Value]] of null, then set |result| to a
+      [=normal completion=] given |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/bid=].

```

Comment:
Two questions here:
 1. What does it mean to set it to a normal completion _given_ something? Do you intend to set the `[[Value]]` of the newly-created normal completion you're creating?
 2. Why does this only happen if this is a throw completion with empty `[[Value]]`? Does this match what the implementation does, and is this tested?

Reply:
> 1. What does it mean to set it to a normal completion _given_ something? 

I meant to pass in the something as the constructor parameter: https://tc39.es/ecma262/#sec-normalcompletion
Similar to how we do so here for throw completion: https://wicg.github.io/turtledove/#:~:text=a%20new%20throw%20completion%20given%20null.

> Do you intend to set the `[[Value]]` of the newly-created normal completion you're creating?

I intended to have the constructor do that for me.

> 2. Why does this only happen if this is a throw completion with empty `[[Value]]`? Does this match what the implementation does, and is this tested?

I did this to catch timeouts where I create a throw completion with empty `[[Value]]` like so: https://wicg.github.io/turtledove/#:~:text=a%20new%20throw%20completion%20given%20null.

Reply:
There's a bit of a collision between Infra's convention for calling algorithms with "given" and TC39's convention of  writing function calls to "[abstract operations](https://tc39.es/ecma262/#sec-abstract-operations)". I would say to use TC39's syntax here:

```suggestion
      [$NormalCompletion$](|global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/bid=]).
```

except that Completion Records need to contain ECMAScript values, not Infra or WebIDL values, so this would imply that you should convert the bid to an ES value. But setBid() just got done converting it to an Infra value, and the caller is about to need to convert it back to an Infra value (and the [caller](https://pr-preview.s3.amazonaws.com/WICG/turtledove/pull/605.html#ref-for-evaluate-a-bidding-script) currently converts it to and from `GenerateBidOutput` an extra time). So really, this algorithm shouldn't be creating Completion records at all, and it should directly return the type its caller wants.

Reply:
I reworked this and I think it avoids the conversions and types align properly.